docs/tutorials/dev_add_task.rst

```diff
@@ -0,0 +1,1405 @@
+  other fields).
+
+* Neither OHC nor BSF have observations to compare with.
+
+* Both OHC and BSF require computing a new field, rather than directly using
+  output from MPAS-Ocean.
+
+On the other hand, there are some major differences between the 2 that will
+mean my job isn't a simple substitution:
+
+* While OHC is computed over different depth ranges, we do not want that for
+  the BSF analysis.
+
+* We will eventually want some "fancier" plotting for the BSF that draws
+  streamlines with arrows.  That's not currently available in any MPAS-Analysis
+  tasks.
+
+* OHC involves computing an anomaly, but that isn't anything we need for BSF.
+
+Even so, :py:class:`~mpas_analysis.ocean.ClimatologyMapOHCAnomaly` seems like
+a reasonable starting point.
+
+4. Developing the task
+----------------------
+
+I'll start just by making a new worktree, then copying the "template" analysis
+task to the new name:
+
+.. code-block:: bash
+
+    git worktree add ../add_climatology_map_bsf
+    cd ../add_climatology_map_bsf
+    cp mpas_analysis/ocean/climatology_map_ohc_anomaly.py mpas_analysis/ocean/climatology_map_bsf.py
+
+Then, I'll open this new worktree in PyCharm.  (You can, of course, use
+whatever editor you like.)
+
+.. code-block:: bash
+
+    pycharm-community .
+
+I'll create or recreate my ``mpas_dev`` environment as in
+:ref:`tutorial_dev_getting_started`, and then make sure to at least do:
+
+.. code-block:: bash
+
+    conda activate mpas_dev
+    python -m pip install -e .
+
+4.1 ``ClimatologyMapBSF`` class
+-------------------------------
+
+In the editor, I rename the class from ``ClimatologyMapOHCAnomaly`` to
+``ClimatologyMapBSF`` and task name from ``climatologyMapOHCAnomaly`` to
+``climatologyMapBSF``.
+
+Then, I update the docstring right away because otherwise I'll forget!
+
+.. code-block:: python
+
+    class ClimatologyMapBSF(AnalysisTask):
+        """
+        An analysis task for computing and plotting maps of the barotropic
+        streamfunction (BSF)
+
+        Attributes
+        ----------
+        mpas_climatology_task : mpas_analysis.shared.climatology.MpasClimatologyTask
+            The task that produced the climatology to be remapped and plotted
+        """
+
+I keep the ``mpas_climatology_task`` attribute because I'm going to need a
+climatology of the velocity field and layer thicknesses that I will get from
+that task, but I know I won't need the ``ref_year_climatology_task`` attribute
+so I get rid of it.
+
+4.2 Constructor
+~~~~~~~~~~~~~~~
+
+Then, I move on to the constructor.  The main things I need to do besides
+renaming the task are:
+
+* rename the field I'm processing to ``barotropicStreamfunction``.
+
+* clean up the ``tags`` a little bit (change ``anomaly`` to ``streamfunction``).
+
+* get rid of ``ref_year_climatology_task`` since I'm not computing anomalies.
+
+* get rid of ``depth_range`` because I'm using only the full ocean column.
```

Comment:
Do we add the depth range back in here?

Reply:
I don't think so.  Let's keep this as simple as possible for now.